src/lib/loadPageMetadataMap.ts

```diff
@@ -22,27 +22,65 @@ export type PageMetadataMapItemWithObjectID = PageMetadataMapItem & {
 
 export type PageMetadataMap = Map<string, PageMetadataMapItem>;
 
-const query = gql`
-  query PageMetadataCollection($preview: Boolean = false) {
-    pageMetadataCollection(limit: 500, preview: $preview) {
-      items {
-        sys {
-          id
+const pageMetadataItems = gql`
+  fragment PageMetadataItems on PageMetadataCollection {
+    items {
+      sys {
+        id
+      }
+      title
+      slug
+      isRoot
+      parent {
+        ... on PageMetadata {
+          sys {
+            id
+          }
         }
-        title
-        slug
-        isRoot
-        parent {
-          ... on PageMetadata {
-            sys {
-              id
-            }
+      }
+      internalRedirect {
```

Comment:
RE: @hinzed1127 separating the query so we only grab redirects, I think that would have been less efficient since we would be making double the requests to Contentful (when we could just grab it all in one go while we're working on loading the PageMetadataMap).

That said, I wanted to create a fragment so we could have two separate queries (one for all the entries and one for only entries that don't include redirects), but in both cases here we're still asking for data on `internalRedirect` and `externalRedirect` even though in one of those cases we know that these will all come back `null`. Unsure how to clean this up so that we don't query for values we know are `null` without removing the fragment and duplicating the majority of the query.

Reply:
> Unsure how to clean this up so that we don't query for values we know are null...

Not sure if I follow this bit. What scenario would do that?

Reply:
The fragment here is querying for `internalRedirect` and `externalRedirect`, but we use the fragment for both types of queries:
1. `queryAll` - we do this when `includeRedirects` is `true` and therefore we want the data on `internalRedirect` and `externalRedirect`
2. `queryWithoutRedirects` - we do this when `includeRedirects` is `false` and we apply a filter:
    ```
    where: { AND: [{ internalRedirect_exists: false }, { externalRedirect_exists: false }] }
    ```
    But because we're reusing the same fragment we'll still get back `internalRedirect` and `externalRedirect` in the response (all of which will be `null` based on the filter).

Ideally we wouldn't query for `internalRedirect` and `externalRedirect` in the second case, but I don't think there's a way to do that without duplicating the GraphQL query?